src/test/librbd/image/test_mock_RemoveRequest.cc

```diff
@@ -0,0 +1,315 @@
+    auto &expect = EXPECT_CALL(get_mock_io_ctx(mock_image_ctx.md_ctx),
+                               list_watchers(mock_image_ctx.header_oid, _));
+    if (r < 0) {
+      expect.WillOnce(Return(r));
+    } else {
+      obj_watch_t watcher;
+      strcpy(watcher.addr, (address + ":0/0").c_str());
+      watcher.cookie = watch_handle;
+
+      std::list<obj_watch_t> watchers;
+      watchers.push_back(watcher);
+
+      expect.WillOnce(DoAll(SetArgPointee<1>(watchers), Return(0)));
+    }
+  }
+
+  void expect_get_group(MockImageCtx &mock_image_ctx, int r) {
+    auto &expect = EXPECT_CALL(get_mock_io_ctx(mock_image_ctx.md_ctx),
+                               exec(mock_image_ctx.header_oid, _, StrEq("rbd"),
+                                    StrEq("image_get_group"), _, _, _));
+    if (r < 0) {
+      expect.WillOnce(Return(r));
+    } else {
+      expect.WillOnce(DoDefault());
+    }
+  }
+
+  void expect_trim(MockImageCtx &mock_image_ctx,
+                   MockTrimRequest &mock_trim_request, int r) {
+    EXPECT_CALL(mock_trim_request, send())
+                  .WillOnce(FinishRequest(&mock_trim_request, r, &mock_image_ctx));
+  }
+
+  void expect_disable_features(MockImageCtx &mock_image_ctx,
+                   MockDisableFeaturesRequest &mock_disable_features_request, int r) {
+    EXPECT_CALL(mock_disable_features_request, send())
+                  .WillOnce(FinishRequest(&mock_disable_features_request, r, &mock_image_ctx));
+  }
+
+  void expect_remove_child(MockImageCtx &mock_image_ctx, int r) {
+    EXPECT_CALL(get_mock_io_ctx(mock_image_ctx.md_ctx),
+                               exec(RBD_CHILDREN, _, StrEq("rbd"), StrEq("remove_child"), _,
+                                    _, _))
+      .WillOnce(Return(r));
+  }
+
+  void expect_remove_mirror_image(MockImageCtx &mock_image_ctx, int r) {
+    EXPECT_CALL(get_mock_io_ctx(mock_image_ctx.md_ctx),
+                exec(RBD_MIRRORING, _, StrEq("rbd"), StrEq("mirror_image_remove"),
+                     _, _, _))
+      .WillOnce(Return(r));
+  }
+
+  void expect_mirror_image_get(MockImageCtx &mock_image_ctx, int r) {
+    EXPECT_CALL(get_mock_io_ctx(mock_image_ctx.md_ctx),
+                exec(RBD_MIRRORING, _, StrEq("rbd"), StrEq("mirror_image_get"),
+                     _, _, _))
+      .WillOnce(Return(r));
+  }
+
+  void expect_dir_remove_image(librados::IoCtx &ioctx, int r) {
+    EXPECT_CALL(get_mock_io_ctx(ioctx),
+                exec(RBD_DIRECTORY, _, StrEq("rbd"), StrEq("dir_remove_image"),
+                     _, _, _))
+      .WillOnce(Return(r));
+  }
+};
+
+TEST_F(TestMockImageRemoveRequest, SuccessV1) {
+  REQUIRE_FORMAT_V1();
+  TestImageRemoveSetUp();
+
+  C_SaferCond ctx;
+  librbd::NoOpProgressContext no_op;
+  MockContextWQ op_work_queue;
+  MockTrimRequest mock_trim_request;
+
+  expect_trim(*m_mock_imctx, mock_trim_request, 0);
+  expect_state_open(*m_mock_imctx, 0);
+  expect_state_close(*m_mock_imctx);
+  expect_op_work_queue(*m_mock_imctx);
```

Comment:
Currently, this isn't providing much value. You might just be better off keeping the direct instantiation of ContextWQ in RemoveRequest and avoid the extra complexities of mocking it.